### PR TITLE
Add filtering support to consul_service and consul_service_health data sources

### PR DESCRIPTION
**Pull request by remilapeyre**

### CODE REVIEW

File: consul/data_source_consul_service_health.go [hunk 1]
```diff
@@ -16,84 +16,81 @@ func dataSourceConsulServiceHealth() *schema.Resource {
...
-				// ForceNew: true,
-				Default: true,
+				Default:  true,
 			},
-			"wait_for": &schema.Schema{
+			"wait_for": {
+				Optional: true,
+				Type:     schema.TypeString,
+			},
+			"filter": {
```
Comment (mkeeler) [inline consul/data_source_consul_service_health.go:51]: Should the filter have `ForceNew: true` or is that not necessary to make terraform make an api request with the new filter?

Comment (mkeeler) [reply consul/data_source_consul_service_health.go:51]: After a little more thought I am fairly certain its not needed and TF will do the right thing as-is. Getting some confirmation on that point would be great though.

Comment (remilapeyre) [reply consul/data_source_consul_service_health.go:51]: Good question, for datasources `ForceNew` is not needed as a request they only implement `Read` they will always make an API call no matter whether their arguments changed. `ForceNew` is only meaningful for resources that implements `Update`, for resources that don't implement `Update` all arguments must have `ForceNew: true`.

**Comment by remilapeyre** [comment]
Thanks for the review @mkeeler !

File: consul/data_source_consul_service_health.go (hunk 1 above)
Comment (mkeeler) [reply consul/data_source_consul_service_health.go:51]: 👍

**Pull request merged**